pkg/asset/installconfig/gcp/client.go

```diff
@@ -80,6 +81,40 @@ func (c *Client) GetMachineType(ctx context.Context, project, zone, machineType
 	return req, nil
 }
 
+// GetMachineTypeWithZones retrieves the specified machine type and the zones in which it is available.
+func (c *Client) GetMachineTypeWithZones(ctx context.Context, project, region, machineType string) (*compute.MachineType, sets.Set[string], error) {
+	svc, err := c.getComputeService(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	var machine *compute.MachineType
+	zones := sets.New[string]()
+	filter := fmt.Sprintf("name = \"%s\" AND zone : %s-*", machineType, region)
+	req := svc.MachineTypes.AggregatedList(project).Filter(filter).Context(ctx)
+	err = req.Pages(ctx, func(page *compute.MachineTypeAggregatedList) error {
+		for _, scopedList := range page.Items {
```

Comment:
i'm guessing there'll be only 1 "MachineTypesScopedList" item because this is just for one instance type right? wondering if we can get away with just referencing the first item ([0])

Reply:
We could technically access it like that yes, but you still need to have checks to ensure that items were returned. I believe that it is still possible to have 0 returned in which a static access would fail.

Reply:
We're still going through the whole list but now because it's a shared code. It should address this comment.